src/audio/component.c

```diff
@@ -182,8 +182,9 @@ void comp_get_copy_limits_frame_aligned(const struct comp_buffer __sparse_cache
 
 #ifdef STREAMCOPY_HIFI3
 
-int audio_stream_copy(const struct audio_stream __sparse_cache *source, uint32_t ioffset,
-		      struct audio_stream __sparse_cache *sink, uint32_t ooffset, uint32_t samples)
+void audio_stream_copy(const struct audio_stream __sparse_cache *source, uint32_t ioffset,
```

Comment:
what about failures here ? i.e. some error occurs in a module during copy. modules should also be allowed to return 0 bytes.